### PR TITLE
feat: add the possibility to show the value over the bar on the Chart…

### PR DESCRIPTION
**Pull request by rgah2107**
fix: #1590

## Changes proposed in this PR:
- add the possibility to show the value over the bar on the Chart component

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/Chart/resolveOptions.js [hunk 1]
```diff
@@ -30,7 +35,7 @@ export default function resolveOptions(conditions) {
     };
 
     let options = {
-        maintainAspectRatio,
+        maintainAspectRatio: maintainAspectRatio,
```
Comment (LeandroTorresSicilia) [inline src/components/Chart/resolveOptions.js:38]: why this change? I think it is better using the object shorthand

File: src/components/Chart/resolveOptions.js [hunk 2]
```diff
@@ -92,15 +105,54 @@ export default function resolveOptions(conditions) {
...
+                                  meta.data.forEach((bar, index) => {
+                                      const dataValue = dataset.data[index];
+                                      ctx.fillText(dataValue, bar._model.x, bar._model.y - 5);
+                                  });
+                              });
+                          }
+                        : null,
+            },
+        };
+    }
```
Comment (LeandroTorresSicilia) [inline src/components/Chart/resolveOptions.js:147]: we should add test cases for this changes

File: src/components/Chart/resolveOptions.js [hunk 3]
```diff
@@ -1,3 +1,7 @@
+/* eslint-disable func-names */
+/* eslint-disable object-shorthand */
+/* eslint-disable no-underscore-dangle */
```
Comment (LeandroTorresSicilia) [inline src/components/Chart/resolveOptions.js:3]: we should avoid adding eslint exceptions, only in some situations, keep in mind that they encourage best practices

File: src/components/Chart/resolveOptions.js [hunk 4]
```diff
@@ -92,15 +105,54 @@ export default function resolveOptions(conditions) {
...
+            hover: {
+                animationDuration: disableAnimations
+                    ? 0
+                    : ChartJS.defaults.global.hover.animationDuration,
+            },
+            animation: {
+                duration: disableAnimations ? 0 : ChartJS.defaults.global.animation.duration,
+                onComplete:
+                    showLabelsOnBars && type === 'bar'
+                        ? function() {
```
Comment (LeandroTorresSicilia) [inline src/components/Chart/resolveOptions.js:123]: instead of using ```function() {}``` here, you can use an arrow function:
```? () => {}```

File: src/components/Chart/resolveOptions.js [hunk 5]
```diff
@@ -92,15 +105,54 @@ export default function resolveOptions(conditions) {
...
+
+                              this.data.datasets.forEach((dataset, i) => {
+                                  const meta = chart.controller.getDatasetMeta(i);
+                                  meta.data.forEach((bar, index) => {
+                                      const dataValue = dataset.data[index];
+                                      ctx.fillText(dataValue, bar._model.x, bar._model.y - 5);
+                                  });
+                              });
+                          }
+                        : null,
```
Comment (LeandroTorresSicilia) [inline src/components/Chart/resolveOptions.js:144]: we should move this complex function to another file in order to test it in isolation

File: src/components/Chart/resolveLabelsOnBars.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable no-underscore-dangle */
+import ChartJS from 'chart.js';
+import defaultTheme from '../../styles/defaultTheme';
+
+const palette = defaultTheme.palette;
```
Comment (yvmunayev) [inline src/components/Chart/resolveLabelsOnBars.js:5]: You can't just use the defaultTheme, I think a possible solution is to pass the parameter palette to this function when you call it in resolveOptions.js

Comment (rgah2107) [reply src/components/Chart/resolveLabelsOnBars.js:5]: I tried that at first but the function loses the instance of the chart when I pass some parameter, however check my next commit, I made a change regarding this, anyway tell me if you see a better solution
